### ISSUE TITLE
Docs: The requested resource does not exist

### ISSUE THREAD

**Issue by Boscop** [Byron/google-apis-rs#213]
I want to use this crate to append rows to a google sheet, but where can I find the docs?
Here it says "The requested resource does not exist":
https://docs.rs/google-sheets4/1.0.8+20181010/google_sheets4/

**Comment by Boscop** [comment]
Also, these docs are also 404:
http://byron.github.io/google-apis-rs/google_sheets4_cli/sheets/api/samples/writing#append_values
http://byron.github.io/google-apis-rs/google_sheets4_cli/sheets/api/guides/values#appending_values
which are linked to from this page:
http://byron.github.io/google-apis-rs/google_sheets4_cli/spreadsheets_values-append/

Btw, I installed the `google-sheets4-cli` but when I invoke it, and open the link that it shows in my browser, I get:
```
401. That’s an error.

Error: deleted_client

The OAuth client was deleted.
```
:/

**Comment by Boscop** [comment]
I want to do it manually from Rust code now, so I need to construct it like here:
https://github.com/Byron/google-apis-rs/blob/bce4f0855481c68027cea5f85b9cb079b70e5a76/gen/sheets4-cli/src/main.rs#L681-L682
but where is `ValueRange` defined?
It doesn't show up here:
https://github.com/Byron/google-apis-rs/search?q=ValueRange&unscoped_q=ValueRange

Do I have to generate the Rust API code myself?

**Comment by Boscop** [comment]
Hm, when I try it now, my request to append values to my spreadsheet fails with:
> Token retrieval failed with error: Invalid Scope: 'no description provided'

Any idea why?

**Comment by Boscop** [comment]
Ah, I figured it out, I had to use `DiskTokenStorage` -.-

But this issue remains: How to pass the `spreadsheets` scope to the CLI?

**Comment by IslandUsurper** [comment]
Let's get back to the original issue: docs fail to build.

It looks like the `Color` struct in sheets4 has examples in different languages, but rustdoc interprets them as Rust code. Possibly because they are indented. As an experiment, removing the indention lets docs get built, but the examples look ugly. Instead, surrounding the examples in their lang-specific markdown, ````java` for example, avoids the issue, and looks better.

Unfortunately, I'm not sure how to make that change through Mako. Are the docs coming directly from Google? Is there a way to make one-off changes like this through the build process?

**Comment by IslandUsurper** [comment]
Let's get back to the original issue: docs fail to build.

It looks like the `Color` struct in sheets4 has examples in different languages, but rustdoc interprets them as Rust code. Possibly because they are indented. As an experiment, removing the indention lets docs get built, but the examples look ugly. Instead, surrounding the examples in their lang-specific markdown, ````java` for example, avoids the issue, and looks better.

Unfortunately, I'm not sure how to make that change through Mako. Are the docs coming directly from Google? Is there a way to make one-off changes like this through the build process?

*Update* Before hitting send, I found the JSON files describing the APIs. Regenerating those will break my fix, but there may need to be a general system for that in any case.

**Comment by IslandUsurper** [comment]
So, I found the JSON files describing the API. I just wonder how often those get regenerated, and how to keep my change merged in.

**Comment by Byron** [comment]
> Ah, I figured it out, I had to use `DiskTokenStorage` -.-
> 
> But this issue remains: How to pass the `spreadsheets` scope to the CLI?

First of all, sorry for the late reply and thanks for all the research you have poured into this issue thus far!

Assuming that you have the official CLI (e.g. by running `cargo install google-sheets4-cli`) already available, you can set the desired/required scope using the `--scope` flag. Based on this, it will try to find a suitable token on disk, or request it via user interaction. If you are trying to write an own CLI, it would have to mimic this kind of logic I presume.

**Comment by Byron** [comment]
> So, I found the JSON files describing the API. I just wonder how often those get regenerated, and how to keep my change merged in.

Right now there is no way to keep it in, but I think the system could be adjusted to look for patch files, and reapply these automatically after updating the corresponding JSON API descriptor.

**Comment by Goncalerta** [comment]
I'd like to point out that, although this issue is closed, the [docs are still not compiling](https://docs.rs/crate/google-sheets4/1.0.8+20181010).

The fix on #214 was reverted on 86a884c48b178228abe841683d6c65869eb91635.

However, the fix was never published on crates.io, being the last version published at `Oct 14, 2018` and #214 was merged at `Dec 21, 2018`.

**Comment by Byron** [comment]
The sheets documentation is now available at https://docs.rs/google-sheets4/1.0.10+20190625/google_sheets4 .

Three additional APIs failed, but have been fixed (permanently) as well: `chat1`, `vision1`, `spanner1`.
